### PR TITLE
Windows: Add file version information

### PR DESCRIPTION
**Pull request by jstarks**
This change adds file version information to docker.exe and dockerd.exe by
adding a Windows version resource with the windres tool.

This change adds a dependency to binutils-mingw-w64 on Linux, but removes
a dependency on rsrc. Most Windows build environments should already have
windres if they have gcc (which is necessary to build dockerd).

This change is not that important, but the change to how resources are generated is necessary for an upcoming change to run dockerd as a service.

cc @jhowardmsft

### CODE REVIEW

**Comment by jstarks** [comment]
gccgo failure seems spurious (it's also hitting in #22265, which is a non-code change)

**Comment by jhowardmsft** [comment]
LGTM.  Nice! Restarted failing CI.

**Comment by jhowardmsft** [comment]
@tiborvass Any chance you can review if you have time. Thx.

**Comment by tiborvass** [comment]
As I said at https://github.com/docker/docker/pull/22276#issuecomment-213960563, I would prefer renaming `client` to `docker` and `docker` to `dockerd`. Apart from that, looks good although I didn't test it on windows but I'm sure @jhowardmsft did :)

**Comment by jstarks** [comment]
I've updated the code with renames of client -> docker and docker -> dockerd in anticipation of #22276 going through.

File: hack/make/.go-autogen [hunk 1]
```diff
@@ -20,44 +20,40 @@ const (
...
+		$WINDRES \
+			-i hack/make/.resources-windows/$1 \
+			-o $3 \
+			-F $2 \
+			--use-temp-file \
+			-I autogen/winresources/tmp \
+			$defs
+	}
+
+	makeres docker.rc pe-x86-64 autogen/winresources/docker/rsrc_amd64.syso
```
Comment (jhowardmsft) [inline hack/make/.go-autogen]: Why are you putting these in docker and dockerd subdirectories - they're common between client and daemon

Comment (jstarks) [inline hack/make/.go-autogen]: In theory I could have a common .syso with the stuff that's the same between them. Unfortunately, the Go linker does not support combining multiple resources together, which means that the client build fails when I do this. (The daemon build uses cgo for SQLite and so uses the GNU linker, which does support combining resources.)

**Comment by jstarks** [comment]
What's going on with DockerSuite.TestRestartContainerwithRestartPolicy in TP5?

**Comment by jhowardmsft** [comment]
Flakey. Under investigation

**Comment by jstarks** [comment]
I have validated that this successfully builds in Linux with GOOS=windows.

I'd really like to get this in ASAP so that I can open a PR to make Docker run as a service.

**Comment by icecrime** [comment]
LGTM (which in this case means: "I'm sure you know what you're doing").

**Pull request merged**